Fix #3224 - conversion for TIME '24:00' to LocalTime breaks in binary-mode

pmenke-de opened the pull request:
### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](https://github.com/pgjdbc/pgjdbc/blob/master/CONTRIBUTING.md) document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes to Existing Features:

* [ ] Does this break existing behaviour? If so please explain.
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Description

Fixes the bug described in #3224

Conversion of binary-values of `TIME` has been adapted to interpret `24:00` as `LocalTime.MAX` - as does conversion from text-values.
As PostgreSQL only supports microsecond-precision, all values larger than `23:59:59.999_999_499` are converted to `LocalTime.MAX` - as is done in other places in TimestampUtils.java

Binary test-values for [TimestampUtilsTest.java](https://github.com/pgjdbc/pgjdbc/compare/master...pmenke-de:pgjdbc:fix-localtime-max?expand=1#diff-97a2717f6b8899b272ea539e9b1dcd0baa5b8e0555172772076cc2ca92f4ce21) are guessed, as I couldn't find a documentation / source, of how they're actually computed.

pmenke-de committed:
test: add tests for `TIME '24:00'` / LocalTime.MAX in binary-mode

add tests which verify, that conversion of binary-transferred TIME/TIMETZ `24:00` to LocalTime currently fails

pmenke-de committed:
fix: handling of `TIME '24:00'` / LocalTime in binary-mode

fixes conversion of TIME/TIMETZ `24:00` to LocalTime from binary.
we now convert `24:00` to `LocalTime.MAX` instead of failing with `Invalid value for NanoOfDay`, which is now consistent with conversion from text.

davecramer commented on pgjdbc/src/test/java/org/postgresql/test/jdbc42/GetObject310Test.java:
```diff
@@ -214,6 +214,25 @@ public void testGetLocalTime() throws SQLException {
+        assertEquals(localTime, rs.getObject("time_without_time_zone_column", LocalTime.class));
+        assertEquals(localTime, rs.getObject(1, LocalTime.class));
+      }
+      stmt.executeUpdate("DELETE FROM table1");
```
No need to delete, as we drop the table after the test

pmenke-de committed:
test: remove superfluous delete at end of test

`#setUp()` and `#tearDown()` manage the test-table's lifecycle. the test-table gets dropped at end of test anyway.

vlsi commented:
After a bit of thought I am leaning towards reverting the fix.
The reason is as follows: `24:00` time is not supported by `java.time`, and `getObject(..., java.time)` should throw instead of corrupting the data.

It is effectively the same as `infinity` with `getBigDecimal` vs `getDouble`.
PostgreSQL's `numeric` support `infinity` while Java's `BigDecimal` does not. So pgjdbc throws an exception for `getBigDecimal` on `infinity` values.
